virttest/qemu_vm.py

```diff
@@ -1493,7 +1493,8 @@ def sort_key(dev):
                 cmd += "numactl -m %s " % n
 
         # Start constructing devices representation
-        devices = qcontainer.DevContainer(qemu_binary, self.name,
+        machine_type = params.get('machine_type').split(':', 1)[-1]
+        devices = qcontainer.DevContainer(qemu_binary, self.name, machine_type,
```

Comment:
Well I like the idea of using the machine-type used in testing when probing qemu, but it would affect negative testing with bad machine types. Have you encountered a situation where `-M virt` was not enough?